internal/broker/broker.go

```diff
@@ -35,23 +37,23 @@ const (
 
 // Config is the configuration for the broker.
 type Config struct {
-	IssuerURL          string
-	ClientID           string
-	CachePath          string
-	HomeBaseDir        string
-	AllowedSSHSuffixes []string
+	ConfigPath            string
+	DataDir               string
+	OldEncryptedTokensDir string
+
+	clientID           string
+	issuerURL          string
+	homeBaseDir        string
+	allowedSSHSuffixes []string
```

Comment:
> I think that the Config struct knowing about the path we use to parse it is an antipattern.

Yeah I'm also not sure how I feel about that. We could move the config path out of the struct and pass it as a separate argument to `broker.New`.

Reply:
Could be an approach, for sure. Not sure how it would like. Doesn't it feel wrong to pass a file path as an argument to the broker creation, though? I think that splitting into two configuration structs (one for what we need to create the broker and one for the broker itself) might be better. But passing `ConfigPath`, `DataDir` and `OldEncryptedTokensDir` as arguments to broker.New also could be good.

Honestly, I have no strong opinion regarding which of the two is the best, so it's your call.

Reply:
> Doesn't it feel wrong to pass a file path as an argument to the broker creation, though?

Doesn't feel wrong to me.

> I think that splitting into two configuration structs (one for what we need to create the broker and one for the broker itself) might be better.

Can you propose sensible names for those two structs? I can't think of any, and I think they serve the same purpose, so I would like to keep it as one struct.

Reply:
The only thing that comes to mind now is `broker.Paths`, which would be the required thing for creating the broker and `broker.Config` which would be the values that we parse from the configuration file. How does it sound? Honest opinion.

Reply:
I could live with it, but it feels wrong that `homeBaseDir` is part of the `broker.Config` struct and not of the `broker.Paths` struct (and the same applies for any other paths which we might want to make configurable in the config file at some point).

Reply:
I can see that. However, the `homeBaseDir` is related to the generated user info rather than the paths used by the broker itself. Maybe we can reflect this in the naming? e.g. `usersHomeBaseDir`

Reply:
I extracted a `userConfig` struct which I embed in the `Config` struct in the latest commit I pushed (295c8713bbcdb0cb6bcb3ebee50841e4235c23ab). What do you think?

Reply:
Seems good to me. Do we gain anything from having it embedded into the Config struct?

Reply:
> Do we gain anything from having it embedded into the Config struct?

The `broker` can access all of its config via `b.cfg`.